include/inexor/vulkan-renderer/settings_decision_maker.hpp

```diff
@@ -72,11 +78,8 @@ struct VulkanSettingsDecisionMaker {
     /// @param surface The selected (window) surface.
     /// @param window_width The width of the window.
     /// @param window_height The height of the window.
-    /// @param swapchain_extent [out] The extent of the swapchain.
-    /// @todo Make this function return a std::optional<VkExtend2D> instead of using call by reference!
-    void decide_swapchain_extent(const VkPhysicalDevice &graphics_card, const VkSurfaceKHR &surface,
-                                 std::uint32_t &window_width, std::uint32_t &window_height,
-                                 VkExtent2D &swapchain_extent);
+    SwapchainSettings decide_swapchain_extent(const VkPhysicalDevice &graphics_card, const VkSurfaceKHR &surface,
```

Comment:
I did this to remove the call by reference arguments. We should get rid of call by value for writing into parameter's memory entirely.